NXP/MCIMX8M_EVK_4GB/Library/MsPlatformDevicesLib/MsPlatformDevicesLib.c

```diff
@@ -0,0 +1,284 @@
+/*++
+
+    Copyright (C) 2019 Microsoft Corporation. All Rights Reserved.
+
+Module Name:
+
+    MsPlatformDevicesLib.c
+
+Abstract:
+
+    This module will provide the device path to the SDCard, the list of
+    devices that must always be connected, and the list of devices that
+    need to be connected for ConIn OnDemand.
+
+--*/
+
+#include <Protocol/DevicePath.h>
+#include <Library/DeviceBootManagerLib.h>
+#include <Library/DevicePathLib.h>
+#include <Library/MsPlatformDevicesLib.h>
+#include <Protocol/GraphicsOutput.h>
+#include <Library/PcdLib.h>
+#include <Library/DebugLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+
+typedef struct {
+  ACPI_HID_DEVICE_PATH      PciRootBridge;
+  PCI_DEVICE_PATH           PciDevice;
+  USB_CLASS_DEVICE_PATH     Usb;
+  EFI_DEVICE_PATH_PROTOCOL  End;
+} GENERIC_USB_CLASS_DEVICE_PATH;
+
+
+#define gEndEntire \
+  { \
+    END_DEVICE_PATH_TYPE, \
+    END_ENTIRE_DEVICE_PATH_SUBTYPE, \
+    { \
+      END_DEVICE_PATH_LENGTH, \
+      0 \
+    } \
+  }
+
+//89464DAE-8DAA-41FE-A4C8-40 D2 17 5A F1 E9
+#define GRAPHICS_DEVICE_PATH_GUID  {0x89464DAE, 0x8DAA, 0x41FE, {0xa4, 0xc8, 0x40, 0xd2, 0x17, 0x5a, 0xf1, 0xe9}}
```

Comment:
Can this be declared as a Pcd element?  It's already defined in two other files.

Reply:
Why is this not done in a DEC file? Does it need to be changed per-platform? If so, yes, PCD is probably best (has the most prior art).